v10 Feature: queryRenderedFeaturesInRect will query visible map if empty array is passed

mysport12 opened the pull request:
## Description

Passing an empty array to queryRenderedFeaturesInRect will now query the visible map view

## Checklist

<!-- Check completed item: [X] -->

- [x] I have tested this on a device/simulator for each compatible OS
- [x] I updated the documentation with running `yarn generate` in the root folder
- [x] I updated the typings files - if js interface was changed (`index.d.ts`)
- [ ] I added/ updated a sample - if a new feature was implemented (`/example`)

mfazekas commented on javascript/components/MapView.js:
```diff
@@ -525,6 +525,31 @@ class MapView extends NativeBridgeComponent(React.Component) {
+   * @param  {Array=} layerIDs -  A array of layer id's to filter the features by
+   * @return {FeatureCollection}
+   */
+  async queryRenderedFeaturesInView(filter = [], layerIDs = []) {
```
Should we add a rect param too?! What do you think?!

mysport12 replied:
No since there is already the queryRenderedFeaturesInRect function. I thought it might be cleaner to separate the two instead of defaulting to entire view if a rect wasn't provided

mfazekas replied:
Yeah, sorry that was my oversight. Yes we already have function that queries with rect in view coordinates.

What I'm concerned about:
- the implemented functionality can be already achieved by `queryRenderedFeaturesInRect` and using `onLayout` to measure view size.
- code duplication: we now have almost duplicate of previous native methods

mfazekas replied:
Yeah, sorry that was my oversight. Yes we already have function that queries with rect in view coordinates.

What I'm concerned about:
- the implemented functionality can be already achieved by `queryRenderedFeaturesInRect` and using `onLayout` to measure view size.
- code duplication: we now have almost duplicate of previous native methods.

mysport12 replied:
Understandable. I can close this PR and will just keep as a local patch instead. For our use case, it is a nice to have helper function so that the layout doesn't have to get measured and then the size passed around along with the mapRef.

mysport12 commented:
@mfazekas reopened with changes. I removed the new function and added the functionality to the existing queryRenderedFeaturesInRect function by passing an empty array instead of the 4 bounding box corners. Let me know if you are ok with the approach.
